keylime/ima/ima.py

```diff
@@ -233,7 +233,8 @@ def _validate_ima_buf(
             failure = _validate_ima_ng(exclude_regex, allowlist, digest, path, hash_types="keyrings")
             if not failure:
                 # Add the key only now that it's validated (no failure)
-                ima_keyrings.add_pubkey_to_keyring(pubkey, path.name, keyidv2=keyidv2)
+                if ima_keyrings is not None:
+                    ima_keyrings.add_pubkey_to_keyring(pubkey, path.name, keyidv2=keyidv2)
```

Comment:
Fixes:   /home/stefanb/keylime/keylime/ima/ima.py:312:90 - error: Argument of type "Unknown | None" cannot be assigned to parameter "ima_keyrings" of type "ImaKeyrings function "_validate_ima_buf" (reportGeneralTypeIssues)

LGTM.